Revit_Core_Engine/Create/View/ViewPlan.cs

```diff
@@ -97,6 +97,77 @@ public static View ViewPlan(this Document document, Autodesk.Revit.DB.Level leve
         }
 
         /***************************************************/
+
+        [Description("Creates and returns a new Floor Plan view in the current Revit file.")]
+        [Input("document", "Revit current document to be processed.")]
+        [Input("level", "The level that the created Floor Plan refers to.")]
+        [Input("viewName", "Optional, name of the new view.")]
+        [Input("scopeBox", "Optional, the Scope Box to attempt to apply to the newly created view.")]
+        [Input("viewTemplateId", "Optional, the View Template Id to be applied in the view.")]
+        [Input("cropRegionVisible", "True if enable crop region visibility.")]
+        [Input("annotationCrop", "True if enable annotation crop visibility.")]
+        [Output("viewPlan", "The new view.")]
+        public static View ViewPlan(this Document document, string viewName, Level level, ElementId scopeBoxId = null, ElementId viewTemplateId = null, bool cropRegionVisible = false, bool annotationCrop = false)
+        {
+            View result = null;
+
+            ViewFamilyType vft = Query.ViewFamilyType(document, ViewFamily.FloorPlan);
+
+            result = Autodesk.Revit.DB.ViewPlan.Create(document, vft.Id, level.Id);
+
+            if (viewTemplateId != null)
+            {
+                try
+                {
+                    result.ViewTemplateId = viewTemplateId;
+                }
+                catch (Exception)
+                {
+                    BH.Engine.Base.Compute.RecordWarning("Could not apply the View Template of Id " + viewTemplateId + "'." + ". Please check if it's a valid ElementId.");
+                }
```

Comment:
It can be cleaner to get the view using its ID, check its `IsTemplate` property, and then do the assignment or raise the warning accordingly.

Reply:
Done 👍